### PR TITLE
Add tradecalender test

### PR DESCRIPTION
**Pull request by Nirus2000**
Only completion of the open test dates of the trading days. No modification or deletion of tests.

The @chirlu for the latest changes and checks.
So that we prevent this I have added these tests.

### CODE REVIEW

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java [hunk 1]
```diff
@@ -159,66 +158,97 @@ public void testMovingHolidays()
...
         // See https://www.nyse.com/markets/hours-calendars
 
         TradeCalendar calendar = TradeCalendarManager.getInstance("nyse");
 
-        assertThat(calendar.isHoliday(LocalDate.parse("2001-09-10")), is(false)); // regular Monday
-        assertThat(calendar.isHoliday(LocalDate.parse("2001-09-11")), is(true)); // closed after terrorist attacks (9/11)
-        assertThat(calendar.isHoliday(LocalDate.parse("2001-09-14")), is(true)); // ... still closed
-        assertThat(calendar.isHoliday(LocalDate.parse("2001-09-17")), is(false)); // reopened the following Monday
-
-        assertThat(calendar.isHoliday(LocalDate.parse("2020-01-01")), is(true));
```
Comment (chirlu) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:173]: Removed despite claiming that no tests were removed.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java [hunk 2]
```diff
@@ -159,66 +158,97 @@ public void testMovingHolidays()
...
-        assertThat(calendar.isHoliday(LocalDate.parse("2020-01-20")), is(true));
```
Comment (chirlu) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:174]: Removed despite claiming that no tests were removed.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java [hunk 3]
```diff
@@ -159,66 +158,97 @@ public void testMovingHolidays()
...
-        assertThat(calendar.isHoliday(LocalDate.parse("2020-02-17")), is(true));
```
Comment (chirlu) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:175]: Removed despite claiming that no tests were removed.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java [hunk 4]
```diff
@@ -159,66 +158,97 @@ public void testMovingHolidays()
...
-        assertThat(calendar.isHoliday(LocalDate.parse("2020-04-10")), is(true));
-        assertThat(calendar.isHoliday(LocalDate.parse("2020-05-25")), is(true));
-        assertThat(calendar.isHoliday(LocalDate.parse("2020-07-03")), is(true));
-        assertThat(calendar.isHoliday(LocalDate.parse("2020-09-07")), is(true));
```
Comment (chirlu) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:179]: Removed despite claiming that no tests were removed.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java [hunk 5]
```diff
@@ -159,66 +158,97 @@ public void testMovingHolidays()
...
-        assertThat(calendar.isHoliday(LocalDate.parse("2020-11-26")), is(true));
```
Comment (chirlu) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:180]: Removed despite claiming that no tests were removed.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java [hunk 6]
```diff
@@ -159,66 +158,97 @@ public void testMovingHolidays()
...
 
     @Test
-    public void testLSE()
+    public void testTradeCalenderLSE()
     {
         // See https://www.londonstockexchange.com/trade/trading-access/business-days
 
         TradeCalendar calendar = TradeCalendarManager.getInstance("lse");
 
-        assertThat(calendar.isHoliday(LocalDate.parse("2019-01-01")), is(true));
```
Comment (chirlu) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:201]: Removed despite claiming that no tests were removed.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java [hunk 7]
```diff
@@ -159,66 +158,97 @@ public void testMovingHolidays()
...
-        assertThat(calendar.isHoliday(LocalDate.parse("2021-12-27")), is(true)); // Monday, substitute for holiday
-        assertThat(calendar.isHoliday(LocalDate.parse("2021-12-28")), is(true)); // Tuesday, substitute for holiday
-        assertThat(calendar.isHoliday(LocalDate.parse("2021-12-29")), is(false));
-        assertThat(calendar.isHoliday(LocalDate.parse("2021-12-30")), is(false));
-        assertThat(calendar.isHoliday(LocalDate.parse("2021-12-31")), is(false));
-        assertThat(calendar.isHoliday(LocalDate.parse("2022-01-01")), is(true)); // Saturday and New Year
-        assertThat(calendar.isHoliday(LocalDate.parse("2022-01-02")), is(true)); // Sunday
-        assertThat(calendar.isHoliday(LocalDate.parse("2022-01-03")), is(true)); // Monday, substitute for holiday
-        assertThat(calendar.isHoliday(LocalDate.parse("2022-01-04")), is(false));
+        // Eastern
```
Comment (chirlu) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:248]: Should be “Easter Monday”. (The whole set of holidays is called Easter, not Eastern, which is a direction.)

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java [hunk 8]
```diff
@@ -251,22 +302,42 @@ public void testLSE()
...
+        // New Year
         assertThat(calendar.isHoliday(LocalDate.parse("2019-01-01")), is(true));
+
+        // Good Friday
         assertThat(calendar.isHoliday(LocalDate.parse("2019-04-19")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2021-04-02")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2022-04-15")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-04-07")), is(true));
+
+        // Eastern
```
Comment (chirlu) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:320]: Same as above.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java [hunk 9]
```diff
@@ -251,22 +302,42 @@ public void testLSE()
...
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-04-10")), is(true));
+
+        // Labour day
         assertThat(calendar.isHoliday(LocalDate.parse("2019-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2020-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2021-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2022-05-01")), is(true));
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-05-01")), is(true));
+
+        // Chrismas
```
Comment (chirlu) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:332]: Missing t.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java [hunk 10]
```diff
@@ -310,6 +416,241 @@ public void testTSX()
...
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-12-25")), is(true));
+
+        // Saint Stephen
+        assertThat(calendar.isHoliday(LocalDate.parse("2023-12-26")), is(true));
+    }
+
+    @Test
+    public void testTradeCalenderMICEXRTS()
+    {
+        // See https://www.wienerborse.at/handel/handelsinformationen/handelskalender/
```
Comment (chirlu) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:605]: Certainly not the place to look for information about Moscow stock exchange.

**Review by chirlu — COMMENTED** [review]
Very annoying PR, especially the claim that “no deletions” were done when there were.

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java (hunk 1 above)
Comment (Nirus2000) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:173]: https://github.com/buchen/portfolio/blob/718f13d64d6f4d6526de484c948a4c4064f393b1/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java#L221
Look here and is not removed

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java (hunk 4 above)
Comment (Nirus2000) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:179]: https://github.com/buchen/portfolio/blob/718f13d64d6f4d6526de484c948a4c4064f393b1/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java#L227
Here... and is not removed

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java (hunk 6 above)
Comment (Nirus2000) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:201]: https://github.com/buchen/portfolio/blob/718f13d64d6f4d6526de484c948a4c4064f393b1/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java#L312
Look here and is not removed

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java (hunk 7 above)
Comment (Nirus2000) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:248]: And before that there was nothing... :-/ 
I have made the effort to go calendrical after the holidays. That there is now Easter and not "Easter Monday" is now hopefully not so bad...

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java (hunk 8 above)
Comment (Nirus2000) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:320]: ...

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java (hunk 9 above)
Comment (Nirus2000) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:332]: ...

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java (hunk 10 above)
Comment (Nirus2000) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/util/TradeCalendarTest.java:605]: thats right... my fault.

**Pull request merged**